### PR TITLE
Get group from `x-kubernetes-group-version-kind` list

### PR DESCRIPTION
**Pull request by rnetser**
## Summary by CodeRabbit

- **New Features**
	- Introduced new `Image` class to manage image import policies and runtime configurations in OpenShift, enhancing control over image stream imports.
	- Added functionality for the `Node` class to manage cluster-wide node information and configuration.
	- Launched new `Project` and `ProjectRequest` classes for managing project configurations and requests, improving project management capabilities in OpenShift.

- **Bug Fixes**
	- Implemented deprecation warnings for removed classes to guide users toward new import paths.

- **Refactor**
	- Streamlined data structures and encapsulated logic within new helper functions for better code maintainability and readability across multiple classes.
	- Consolidated resource validation logic into a new `validate_resource` function, enhancing clarity and efficiency.

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: ocp_resources/project_config_openshift_io.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+              specified, a default template is used.
+
+        """
+        super().__init__(**kwargs)
+
+        self.project_request_message = project_request_message
+        self.project_request_template = project_request_template
+
+    def to_dict(self) -> None:
+        super().to_dict()
```
[the remark this answers is not in the document]
Comment (rnetser) [reply ocp_resources/project_config_openshift_io.py:38]: this is not needed in to_dict

File: ocp_resources/project_project_openshift_io.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            finalizers(List[Any]): Finalizers is an opaque list of values that must be empty to
+              permanently remove object from storage
+
+        """
+        super().__init__(**kwargs)
+
+        self.finalizers = finalizers
+
+    def to_dict(self) -> None:
+        super().to_dict()
```
[the remark this answers is not in the document]
Comment (rnetser) [reply ocp_resources/project_project_openshift_io.py:39]: this is not needed in to_dict

**Review by dbasunag — DISMISSED** [review]

**Review by myakove — APPROVED** [review]

**Pull request merged**